### PR TITLE
Functional contact

### PR DESCRIPTION
**Pull request by PabloFLPs**
Making Contact section finally functional with my Send Email API. Also, I made some changes with the toast emitter, now it is using the "toast.promise()" to show a message while the email is being sended. I don't know if it needs the try/catch block, give me a hint if you know something about this @Erickinhou.

### CODE REVIEW

File: src/components/Contact.tsx [hunk 1]
```diff
@@ -18,7 +18,53 @@
...
+            render(){return '🐈‍⬛ Email Sent!'}
+          },
+          error: {
+            render(){return '🐈‍⬛ Error!'}
+          }
+        }
+      )
+    } catch (error) {
+      //if(error) errorToastEmitter()
+    }
```
Comment (Erickinhou) [inline src/components/Contact.tsx]: I think try catch it's not necessary. But I don't have 100% of certainty.

File: src/components/Contact.tsx [hunk 2]
```diff
@@ -18,7 +18,53 @@
...
+  async function onSubmit(event: any) {
+    event.preventDefault()
+
+    const formToBeSubmitted = {
+      name: name,
+      email: email,
+      message: message + "\n\nPhone: " + phone
+    }
+
+    setDisableButtonState(true)
```
Comment (Erickinhou) [inline src/components/Contact.tsx:44]: disable button it's not working.

**Review by Erickinhou — CHANGES_REQUESTED** [review]
- Add a way to bock the server if I send too much request
- Fix disable button

File: src/components/Contact.tsx (hunk 1 above)
Comment (PabloFLPs) [reply src/components/Contact.tsx]: I just tested it here, seems like it is necessary to catch the error. The button keeps disabled if an error is returned withou the try/catch block.

File: src/components/Contact.tsx (hunk 2 above)
Comment (PabloFLPs) [reply src/components/Contact.tsx:44]: Fixed. I just added an "await" before the toast.promise() function.

**Comment by PabloFLPs** [comment]
@Erickinhou, what exactly is "block the server if too much requests are sended"? It would be on the API or inside the Contact section?

File: src/components/Contact.tsx (hunk 1 above)
Comment (Erickinhou) [reply src/components/Contact.tsx]: okay

**Comment by Erickinhou** [comment]
> @Erickinhou, what exactly is "block the server if too much requests are sended"? It would be on the API or inside the Contact section?

it's better at api.

**Review by Erickinhou — APPROVED** [review]

**Comment by PabloFLPs** [comment]
> > @Erickinhou, what exactly is "block the server if too much requests are sended"? It would be on the API or inside the Contact section?
> 
> it's better at api.

Cool, I will search about it on Heroku.

**Pull request merged**